src/Messages/Interactions/TSInteraction.h

```diff
@@ -33,7 +32,18 @@ NS_ASSUME_NONNULL_BEGIN
 + (instancetype)interactionForTimestamp:(uint64_t)timestamp
                         withTransaction:(YapDatabaseReadWriteTransaction *)transaction;
 
-- (nullable NSDate *)receiptDateForSorting;
+// NSDate has second precision, uint64_t timestamps have millisecond precision
+// so prefer timestampForSorting over dateForSorting.
+- (NSDate *)dateForSorting;
```

Comment:
> carefully constructing the timestamps 

This is the key argument.  It's nice to use really small `theta` (aka increment) values to order the interactions, and it's important that we don't have to worry about whether values are getting ceil()'d, round()'d or floor()'d as they move back and forth between `NSDate` and `uint64_t`.